org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/RemoteCommand.java

```diff
@@ -38,63 +69,392 @@ public RemoteCommand() {
+			// happened...
+			if (connection.getChannel().isClosed() && exitValue != 0) {
+				logger.error("Connection was closed before job was finished, failed " + exitValue);
+				return CommandStatus.FAILED;
+			}
+		}
+
+		// If the job finished correctly, return success. Otherwise return failure
+		if (exitValue == 0)
+			return CommandStatus.SUCCESS;
+		else
+			return CommandStatus.FAILED;
 	}
 
 	/**
-	 * See {@link org.eclipse.ice.commands.Command#run()}
+	 * This function loops over the various commands to be run on the JSch
+	 * connection. Since they can't be run all together they have to be run
+	 * individually.
+	 * 
+	 * @return
 	 */
 	@Override
-	protected CommandStatus run() {
+	protected CommandStatus processJob() {
 
-		return status;
+		// Setup the list of all of the commands that will be launched. JSch can not
+		// launch multiple commands at once, so we need to take the splitCommand and
+		// cd to the correct working directory in front and then actually perform
+		// the execution. The commands are then split by the semi-colon
+		ArrayList<String> completeCommands = new ArrayList<String>();
+		for (String i : commandConfig.getSplitCommand()) {
+			completeCommands.add("cd " + commandConfig.getRemoteWorkingDirectory() + "; " + i);
+		}
+
+		// Now loop over all commands and run them via JSch
+		for (int i = 0; i < completeCommands.size(); i++) {
+			// Open the channel for the executable to be run on
+			try {
+				connection.setChannel(connection.getSession().openChannel("exec"));
+			} catch (JSchException e) {
+				// If it can't be opened, puke
+				e.printStackTrace();
+			}
+
+			// Give the command to the channel connection
+			String thisCommand = completeCommands.get(i);
+			((ChannelExec) connection.getChannel()).setCommand(thisCommand);
+
+			logger.info("Executing command: " + thisCommand + " remotely in the working direcotry "
+					+ commandConfig.getRemoteWorkingDirectory());
+
+			// Set up the input stream
+			connection.getChannel().setInputStream(null);
+			try {
+				// Set the input stream for the connection object
+				connection.setInputStream(connection.getChannel().getInputStream());
+			} catch (IOException e) {
+				// If we can't set the input stream, puke
+				e.printStackTrace();
+			}
+
+			// Setup the output streams and pass them to the connection channel
+			try {
+				stdErrStream = new FileOutputStream(commandConfig.getErrFileName(), true);
+				stdOutStream = new FileOutputStream(commandConfig.getOutFileName(), true);
+				BufferedOutputStream stdOutBufferedStream = new BufferedOutputStream(stdOutStream);
+				// Give the streams to the channel now that their names are appropriately set
+				connection.getChannel().setOutputStream(stdOutBufferedStream);
+				((ChannelExec) connection.getChannel()).setErrStream(stdErrStream);
+			} catch (FileNotFoundException e) {
+				// If logging streams can't be set, puke
+				e.printStackTrace();
```

Comment:
The exception needs to be dealt with or passed up.

Reply:
I've looked through all classes to make sure that all exceptions are properly dealt with.